osutil/group.go

```diff
@@ -79,7 +79,22 @@ func getent(database, name string) (uint64, error) {
 var findUidNoGetentFallback = func(username string) (uint64, error) {
 	myuser, err := user.Lookup(username)
 	if err != nil {
-		return 0, err
+		// XXX: when using cgo, sometimes getpwnam_r will return ESRCH, which
+		// is not currently recognized as an UnknownUserError type of error by
+		// golang's implementation of user.LookupGroup, so to work around that,
+		// we treat all non-nil errors here as UnknownUserError, as actually
+		// the man page for getpwnam_r says:
+		//
+		// > It does not call "not found" an error, hence does not specify what
```

Comment:
we probably want something like

> It [POSIX.1-2001] ...

otherwise it's not clear what it is in that sentence copied out of the man page

Reply:
sure, would prefer to handle in a followup as this is all green and is thus mergable :smile:

Reply:
actually needed more updates here, so this is done